pypeit/spectrographs/spectrograph.py

```diff
@@ -353,39 +409,52 @@ def get_datasec_img(self, filename, det=1, force=True):
+            section (str):  'datasec' or 'oscansec'
             det (int):
                 Detector number (1-indexed)
-            force (:obj:`bool`, optional):
-                Force the image to be remade
 
         Returns:
             `numpy.ndarray`: Integer array identifying the amplifier
             used to read each pixel.
         """
-        if self.datasec_img is None or force:
-            # Check the detector is defined
-            self._check_detector()
-            # Get the image shape
-            raw_naxis = self.get_raw_image_shape(filename, det=det)
-
-            # This *always* returns spectral then spatial
-            binning = self.get_meta_value(filename, 'binning')
-
-            # This *always* returns spectral then spatial
-            data_sections, one_indexed, include_end \
-                    = self.get_image_section(filename, det, section='datasec')
-
-            # Initialize the image (0 means no amplifier)
-            self.datasec_img = np.zeros(raw_naxis, dtype=int)
-            for i in range(self.detector[det-1]['numamplifiers']):
-                # Convert the data section from a string to a slice
-                datasec = parse.sec2slice(data_sections[i], one_indexed=one_indexed,
-                                          include_end=include_end, require_dim=2,
-                                          binning=binning)
-                # Assign the amplifier
-                self.datasec_img[datasec] = i+1
-
-        return self.datasec_img
+        # Check the detector is defined
+        self._check_detector()
+        # Get the image shape
+        raw_naxis = self.get_raw_image_shape(filename, det=det)
+
+        binning_pypeit = self.get_meta_value(filename, 'binning')
+
+        data_sections, one_indexed, include_end \
+                    = self.get_image_section(filename, det, section=section)
+        # Note on data format
+        #--------------------
+        # binning_pypeit = the binning  in the PypeIt convention of (spec, spat)
+        # binning_raw = the binning in the format of the raw data.
+        # In other words: PypeIt requires spec to be the first dimension of the image as read into python. If the
+        # files are stored the other way with spat as the first dimension (as read into python), then the transpose
+        # flag manages this, which is basically the value of the self.detector[det-1]['specaxis'] above.
+        # (Note also that BTW the python convention of storing images is transposed relative to the fits convention
+        # and the datasec typically written to headers. However this flip is dealt with explicitly in the
+        # parse.spec2slice code and is NOT the transpose we are describing and flipping here).
+        # TODO Add a blurb on the PypeIt data model.
+
+        #if transpose:
+        #   binning_raw = (',').join(binning_pypeit.split(',')[::-1])
+        #else:
+        binning_raw = binning_pypeit
```

Comment:
why is this not just called `binning_raw` from the get-go

Reply:
modified